src/module/actor/lancer-actor.ts

```diff
@@ -1110,6 +1169,53 @@ export class LancerActor extends Actor {
   is_deployable(): this is LancerActor & { data: LancerActorDataProperties<EntryType.DEPLOYABLE> } {
     return this.data.type === EntryType.DEPLOYABLE;
   }
+
+
+  /**
+   * Taking a new and old frame/class, swaps the actor and/or token images if 
+   * we detect that the image isn't custom. Will check each individually
+   * @param robot     A MM Mech or NPC, passed through to avoid data overwrites 
+   * @param oldFrame  Old Frame or NPC Class
+   * @param newFrame  New Frame or NPC Class
+   * @returns         The newFrame if any updates were performed
+   */
+  async swapFrameImage(robot: Mech | Npc, oldFrame: Frame | NpcClass | null, newFrame: Frame | NpcClass): Promise<string> {
+    let oldFramePath = frameToPath[oldFrame?.Name || ""];
+    let newFramePath = frameToPath[newFrame?.Name || ""];
+    let defaultImg = robot.Type == EntryType.MECH ? "systems/lancer/assets/icons/mech.svg" : "systems/lancer/assets/icons/npc_class.svg";
+    
+    if(!newFramePath) newFramePath = defaultImg;
+    let changed = false;
+    let newData: any = {}
+
+    // Check the token
+    // Add manual check for the aws images
+    if(this.data.token.img == oldFramePath || this.data.token.img == defaultImg || this.data.token.img?.includes("compcon-image-assets")) {
+      newData.token = {"img": newFramePath};
+      changed = true;
+    }
+    
+    // Check the actor
+    // Add manual check for the aws images
+    if(this.data.img == oldFramePath || this.data.img == defaultImg || this.data.img?.includes("compcon-image-assets")) {
+      newData.img = newFramePath;
+      
+      // Have to set our top level data in MM or it will overwrite it...
+      robot.Flags.top_level_data.img = newFramePath;
+      robot.Flags.top_level_data["token.img"] = newFramePath;
+      changed = true;
+    }
+
+    let a = undefined;
+    let b = undefined;
+
+    if (changed) {
+      console.log(`${lp} Automatically updating image: `, newData);
+      a = await this.update(newData);
```

Comment:
Just `await this.update(newData)` if you just want to block on it. Also lets you kill a/b.

Reply:
And... forgot I left that in for debugging